openstack/client.go

```diff
@@ -363,7 +363,11 @@ func initClientOpts(client *gophercloud.ProviderClient, eo gophercloud.EndpointO
 // NewBareMetalV1 creates a ServiceClient that may be used with the v1
 // bare metal package.
 func NewBareMetalV1(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error) {
-	return initClientOpts(client, eo, "baremetal")
+	sc, err := initClientOpts(client, eo, "baremetal")
+	if !strings.HasSuffix(sc.Endpoint, "v1/") {
```

Comment:
Is it possible that we have suffix `v1` without the `/`?

Reply:
looks like others services have the / at the end like network : 
https://github.com/gophercloud/gophercloud/pull/2486/files/535d1003686b256713d5a47eb4cfd97e90c0fb73#diff-5643779347824e0ec4258ef187be1f23c3b1eeca504fcfe48e47efe0e051b4bfR395

Reply:
yes you are right maybe we nee double check v1/ and v1 ?

Reply:
Maybe it is best to check both ?

```
if !strings.HasSuffix(strings.TrimSuffix(sc.Endpoint, "/"), "v1")
```

Reply:
Looking at the initialization for the other clients, it seems unlikely the suffix has `v1` without the `/`, but better be safe than sorry.